### ISSUE TITLE
nixpkgs-unstable for polkadot broken

### ISSUE THREAD

**Issue by th7nder** [andresilva/polkadot.nix#31]
Hey, I didn't know where to create the issue so it's visible between all of the stuff, so I'm creating it here.

After using [latest ](https://github.com/NixOS/nixpkgs/commit/88195a94f390381c6afcdaa933c2f6ff93959cb4) `nixpkgs-unstable`, basically doing `nix flake update` (https://github.com/eigerco/polka-storage/blob/develop/flake.nix):

```
   Compiling xcm-runtime-apis v0.1.0 (/build/source/polkadot/xcm/xcm-runtime-apis)
   Compiling pallet-xcm v7.0.0 (/build/source/polkadot/xcm/pallet-xcm)
   Compiling polkadot-runtime-common v7.0.0 (/build/source/polkadot/runtime/common)
   Compiling rococo-runtime-constants v7.0.0 (/build/source/polkadot/runtime/rococo/constants)
   Compiling westend-runtime-constants v7.0.0 (/build/source/polkadot/runtime/westend/constants)
error: failed to run custom build command for `rococo-runtime v7.0.0 (/build/source/polkadot/runtime/rococo)`

Caused by:
  process didn't exit successfully: `/build/source/target/production/build/rococo-runtime-8c2f94674cb75ed2/build-script-build` (exit status: 1)
  --- stdout
  Information that should be included in a bug report.
  Executing build command: env -u CARGO_ENCODED_RUSTFLAGS -u RUSTC CARGO_MAKEFLAGS="-j --jobserver-fds=5,6 --jobserver-auth=5,6" CARGO_TARGET_DIR="/build/source/target/x86_64-unknown-linux-gnu/production/wbuild/rococo-runtime/target" RUSTFLAGS="-C target-cpu=mvp -C target-feature=-sign-ext -C link-arg=--export-table -Clink-arg=--export=__heap_base -C link-arg=--import-memory  --cfg substrate_runtime " SKIP_WASM_BUILD="" "/nix/store/q2p507l5rmfw7c4n5pr6gcd5p2gyws2c-cargo-1.83.0/bin/.cargo-wrapped" "rustc" "--target=wasm32-unknown-unknown" "--manifest-path=/build/source/target/x86_64-unknown-linux-gnu/production/wbuild/rococo-runtime/Cargo.toml" "--color=always" "--profile" "production"
  Using rustc version: rustc 1.83.0 (90b35a623 2024-11-26) (built from a source tarball)

  Compacted wasm in 4.417535622s
  Compressed blob in 156.77864ms
  Failed to deserialize `/build/source/target/x86_64-unknown-linux-gnu/production/wbuild/rococo-runtime/rococo_runtime.wasm`: InvalidTableReference(128)

  --- stderr

```
```
warning: build failed, waiting for other jobs to finish...
error: failed to run custom build command for `westend-runtime v7.0.0 (/build/source/polkadot/runtime/westend)`

Caused by:
  process didn't exit successfully: `/build/source/target/production/build/westend-runtime-b4a98672da4c4206/build-script-build` (exit status: 1)
  --- stdout
  Information that should be included in a bug report.
  Executing build command: env -u CARGO_ENCODED_RUSTFLAGS -u RUSTC CARGO_MAKEFLAGS="-j --jobserver-fds=5,6 --jobserver-auth=5,6" CARGO_TARGET_DIR="/build/source/target/x86_64-unknown-linux-gnu/production/wbuild/westend-runtime/target" RUSTFLAGS="-C target-cpu=mvp -C target-feature=-sign-ext -C link-arg=--export-table -Clink-arg=--export=__heap_base -C link-arg=--import-memory  --cfg substrate_runtime " SKIP_WASM_BUILD="" "/nix/store/q2p507l5rmfw7c4n5pr6gcd5p2gyws2c-cargo-1.83.0/bin/.cargo-wrapped" "rustc" "--target=wasm32-unknown-unknown" "--manifest-path=/build/source/target/x86_64-unknown-linux-gnu/production/wbuild/westend-runtime/Cargo.toml" "--color=always" "--profile" "production"
  Using rustc version: rustc 1.83.0 (90b35a623 2024-11-26) (built from a source tarball)

  Compacted wasm in 4.240879655s
  Compressed blob in 164.960222ms
  Failed to deserialize `/build/source/target/x86_64-unknown-linux-gnu/production/wbuild/westend-runtime/westend_runtime.wasm`: InvalidTableReference(128)
```

Is there anything I can do to fix that?

**Comment by andresilva** [comment]
Update polkadot.nix, this was fixed in 4298e53b4c84f58c63e74bf78fbbec1e9f972b78.

**Comment by andresilva** [comment]
Ah, you are using polkadot from nixpkgs (and not from this flake). Either way I already pushed the same fix to nixpkgs should reach unstable soon https://nixpk.gs/pr-tracker.html?pr=369180
